src/diffusers/pipelines/kandinsky/pipeline_kandinsky.py

```diff
@@ -390,8 +408,13 @@ def __call__(
         image_embeds = torch.cat([negative_image_embeds, image_embeds], dim=0).to(
             dtype=prompt_embeds.dtype, device=device
         )
+        if hasattr(self.scheduler, "custom_timesteps"):
```

Comment:
Hmm I don't think this logic makes much sense thb. If we need a different spacing for Kandinsky, let's add this to the DDPMScheduler similarly to: https://github.com/huggingface/diffusers/blob/77f9137f103b04f137b6c487814ea58599ebc200/src/diffusers/schedulers/scheduling_ddim.py#L306

Reply:
@patrickvonplaten 
looked into this, I think kandinsky's method of calculating step ratio is just **slightly** different from what's already in DDPM, maybe we don't need to make any changes as long as it does not affect generation quality? 

cc @williamberman because Kandinsky timesteps is is exactly same as unclip 

- kandinsky/ unclip scheduler
```python
step_ratio = (self.scheduler.config.num_train_timesteps - 1) / (num_inference_steps - 1)
timesteps = (np.arange(0, num_inference_steps) * step_ratio).round()[::-1].copy().astype(np.int64)
```

- ddpm 
```
step_ratio = self.config.num_train_timesteps // self.num_inference_steps
timesteps = (np.arange(0, num_inference_steps) * step_ratio).round()[::-1].copy().astype(np.int64)
```
compared the cheeseburger monster, the one above is generated with Kandinsky timesteps; the one below is generated with ddpm scheduler without any change. I can see the eyes and mouth are a little bit different for the monster on the right but it really doesn't affect quality

![yiyi_test_pipe_ddpm_burger_out_original](https://github.com/huggingface/diffusers/assets/12631849/3341a017-5b25-453b-92dd-fca766d22fcd)

![yiyi_test_pipe_ddpm_burger_out](https://github.com/huggingface/diffusers/assets/12631849/8077c666-47e8-46c5-90fe-4907c1379a45)

only super minor difference here too

![yiyi_test_pipe_ddpm_hair_out_original](https://github.com/huggingface/diffusers/assets/12631849/fa125ae6-f07f-4b24-83a9-7705d918b297)

    
![yiyi_test_pipe_ddpm_hair_out](https://github.com/huggingface/diffusers/assets/12631849/548e9e52-9d2d-414f-af44-3915b367ee18)